src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/IConfigurationProvider.cs

```diff
@@ -17,14 +18,14 @@ public interface IConfigurationProvider
         /// <param name="key">The key.</param>
         /// <param name="value">The value.</param>
         /// <returns><c>True</c> if a value for the specified key was found, otherwise <c>false</c>.</returns>
-        bool TryGet(string key, out string value);
+        bool TryGet(string key, [MaybeNullWhen(false)] out string value);
```

Comment:
I think it would be more consistent (i.e. with dictionary) for this to be nullable with `NotNullWhen(true)`

Reply:
also since you allow assigning null in the `Set` should this just always be nullable? (assuming assigning false doesn't mean delete)

Reply:
Good catch! Configuration values can be `null`, so `TryGet` can return `null` even if value exists.